File: src/main/java/com/skillcentric/visibility/two/ChildTwo.java
package com.skillcentric.visibility.two;


import com.skillcentric.visibility.one.ParentJava;

public class ChildTwo extends ParentJava {

    public void demo() {

        ParentJava parent = new ParentJava();

        parent.publicMethod();

//        parent.protectedMethod();
//        parent.packagePrivateMethod();
    }

    @Override
    public String publicMethod() {
        return super.publicMethod();
    }

    @Override
    protected String protectedMethod() {
        return super.protectedMethod();
    }
}

File: src/main/kotlin/com/skillcentric/visibility/two/childTwo.kt
package com.skillcentric.visibility.two

import com.skillcentric.visibility.one.Parent

class Child : Parent() {

    fun demo() {

        val parent = Parent()

        parent.publicMethod()
        parent.internalMethod()

//        parent.protectedMethod()
//        parent.privateMethod()
    }

    override fun publicMethod() {
        super.publicMethod()
        println("Some implementation")
    }

    override fun internalMethod() {
        super.internalMethod()
        println("Some implementation")
    }

    override fun protectedMethod() {
        super.protectedMethod()
        println("Some implementation")
    }
}